samples/embedded-map/src/components/EmbeddedMap/EmbeddedMap.tsx

```diff
@@ -1,53 +1,84 @@
+
 // Import the necessary CSS for the Mapillary viewer to be styled correctly.
 import "mapillary-js/dist/mapillary.min.css";
 import EmbeddedMapModel from "./EmbeddedMapModel";
+import "./EmbeddedMap.css";
+import { useWatchAndRerender } from "@vertigis/web/ui/hooks";
+
+declare const ResizeObserver;
 
 export default function EmbeddedMap(
     props: LayoutElementProperties<EmbeddedMapModel>
 ): React.ReactElement {
     const { model } = props;
     const mlyRootEl = useRef<HTMLDivElement>();
 
+    const onSyncToggle = () => {
+        model.sync = !model.sync;
+    };
+
+    const onRecenter = () => {
+        void model.recenter();
+    };
+
+    useWatchAndRerender(model, "sync");
+
     useEffect(() => {
-        const mapillary = new Viewer(
-            mlyRootEl.current,
-            model.mapillaryKey,
-            // Mapillary node to start on.
-            "gLV8Jn5A6b6rbVRy2xhkMA",
-            {
-                component: {
-                    // Initialize the view immediately without user interaction.
-                    cover: false,
-                },
-            }
-        );
+        const mapillary = new Viewer(mlyRootEl.current, model.mapillaryKey);
+        mapillary.setTransitionMode(TransitionMode.Instantaneous);
         model.mapillary = mapillary;
 
-        const handleWindowResize = () => {
+        const handleViewportResize = () => {
             mapillary.resize();
         };
 
-        // Viewer size is dynamic so resize should be called every time the window size changes.
-        window.addEventListener("resize", handleWindowResize);
+        // Viewer size is dynamic so resize should be called every time the viewport size changes.
+        const resizeObserver = new ResizeObserver(handleViewportResize);
+        const viewportDiv = mlyRootEl.current; // mlyRootEl.current.parentElement;
+        resizeObserver.observe(viewportDiv);
 
         // Clean up when this component is unmounted from the DOM.
         return () => {
-            window.removeEventListener("resize", handleWindowResize);
+            resizeObserver.unobserve(viewportDiv);
             // Clear out the Mapillary instance property. This will take care of
             // cleaning up.
             model.mapillary = undefined;
         };
     }, [model, model.id, model.mapillaryKey]);
 
     return (
-        <LayoutElement {...props} stretch>
-            <div ref={mlyRootEl} className="EmbeddedMap-map-container"></div>
-        </LayoutElement>
+        <div className={"gcx-component gcx-stretchy"}>
+            <div className="EmbeddedMap-ui-container">
+                <IconButton
+                    className={clsx(
+                        "EmbeddedMap-button",
+                        "EmbeddedMap-sync-button",
+                        { selected: model.sync }
+                    )}
+                    onClick={onSyncToggle}
+                >
+                    <Sync color={model.sync ? "primary" : "secondary"} />
```

Comment:
Web does not have any concept of a secondary color in its branding, so I think it might be best to avoid using it in the sample, since Clara has no control over it.